packages/definitions-parser/src/git.ts

```diff
@@ -28,20 +27,31 @@ Actions runs:
 
 If editing this code, be sure to test on both full and shallow clones.
 */
-export async function gitDiff(log: Logger, definitelyTypedPath: string): Promise<GitDiff[]> {
-  try {
-    await run("git", ["rev-parse", "--verify", sourceBranch]);
-    // If this succeeds, we got the full clone.
-  } catch (_) {
-    // This is a shallow clone.
-    await run("git", ["fetch", sourceRemote, sourceBranch]);
-    await run("git", ["branch", sourceBranch, "FETCH_HEAD"]);
+export async function gitDiff(
+  log: Logger,
+  definitelyTypedPath: string,
+  noFetch: boolean,
+  diffBase: string,
+): Promise<GitDiff[]> {
+  if (!noFetch) {
```

Comment:
Honestly, I don't think we should ever fetch at all. In CI, we'll have fetched everything we need already. Locally, fetching may mean using the wrong ref. I think I actually just want to remove this altogether and make this code purely read-only.